src/main/drivers/flash.c

```diff
@@ -259,4 +274,90 @@ const flashGeometry_t *flashGetGeometry(void)
 
     return &noFlashGeometry;
 }
+
+/*
+ * Flash partitioning
+ *
+ * Partition table is not currently stored on the flash, in-memory only.
+ *
+ * Partitions are required so that Badblock management (inc spare blocks), FlashFS (Blackbox Logging), Configuration and Firmware can be kept separate and tracked.
+ *
+ * Currently, to keep things simple (and working), the following rules apply:
+ *
+ * 1) order of partitions in the paritions table strictly defined as follows
+ *
+ * BAD BLOCK MANAGEMENT
```

Comment:
Looking at the code, it appears to me that the order (from the start of flash) is reversed to what this list shows. Maybe the list should be reversed to make it easier to understand the code, or it should be annotated to make it clear that this is from the end of flash?